lib/features/children/generosity_challenge/pages/generosity_challenge_day_details.dart

```diff
@@ -20,79 +23,109 @@ class GenerosityChallengeDayDetails extends StatelessWidget {
+                Padding(
+                  padding: const EdgeInsets.only(bottom: 5),
+                  child: TextButton.icon(
+                    onPressed: () {
+                      AnalyticsHelper.logEvent(
+                        eventName: AmplitudeEvents
+                            .generosityChallengeDayUndoCompleting,
+                        eventProperties: {
+                          'day': challenge.state.detailedDayIndex + 1,
+                        },
+                      );
+                      challenge
+                          .undoCompletedDay(challenge.state.detailedDayIndex);
                     },
-                  );
-                  challenge.undoCompletedDay(challenge.state.detailedDayIndex);
-                },
-                icon: const Icon(
-                  FontAwesomeIcons.rotateLeft,
-                  size: 20,
-                  color: AppTheme.givtGreen40,
+                    icon: const Icon(
+                      FontAwesomeIcons.rotateLeft,
+                      size: 20,
+                      color: AppTheme.givtGreen40,
+                    ),
+                    label: Text(
+                      'Undo completing',
+                      style: Theme.of(context).textTheme.titleMedium?.copyWith(
+                            color: AppTheme.givtGreen40,
+                            fontFamily: 'Rouna',
+                            fontWeight: FontWeight.w700,
+                          ),
+                    ),
+                  ),
                 ),
-                label: Text(
-                  'Undo completing',
-                  style: Theme.of(context).textTheme.titleMedium?.copyWith(
-                        color: AppTheme.givtGreen40,
-                        fontFamily: 'Rouna',
-                        fontWeight: FontWeight.w700,
-                      ),
+              if (!isDualCard ||
+                  state.status == DailyAssignmentStatus.confirm ||
+                  state.status == DailyAssignmentStatus.completed ||
+                  day.isCompleted)
+                GivtElevatedButton(
+                  onTap: () async {
+                    await showDialog<void>(
+                      context: context,
+                      barrierDismissible: false,
+                      barrierColor: Theme.of(context)
+                          .colorScheme
+                          .primary
+                          .withOpacity(.25),
+                      builder: (context) => FeedbackBannerDialog(task: task),
+                    );
+
+                    await AnalyticsHelper.logEvent(
+                      eventName:
+                          AmplitudeEvents.generosityChallengeDayCompleted,
+                      eventProperties: {
+                        'day': challenge.state.detailedDayIndex + 1,
+                      },
+                    );
+                    await challenge.completeActiveDay();
```

Comment:
at this point I wanna ask why don't we use GenerosityChallengeCubit for it? Isn't DailyAssignmentCubit look redundant?

Reply:
I wanted to use smaller, more 'focused', but youu are right, one more GenerosityChallengeStatus would work as well

Reply:
done